pyproject.toml

```
@@ -0,0 +1,55 @@
+[build-system]
+requires = ["setuptools~=62.6", "wheel~=0.37.1"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name        = "astroid"
+license     = {text = "LGPL-2.1-or-later"}
+description = "An abstract syntax tree for Python with inference support."
+readme      = "README.rst"
+authors     = [
+    {name = "Python Code Quality Authority", email = "code-quality@python.org"}
+]
+keywords    = ["static code analysis", "python", "abstract syntax tree"]
+classifiers = [
+    "Development Status :: 6 - Mature",
+    "Environment :: Console",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: GNU Lesser General Public License v2 (LGPLv2)",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
```

Should we add `3.11` as well?

---

Hmm nice catch, I think we forgot to add it in https://github.com/PyCQA/astroid/pull/1584/

---

Hmm nice catch, I think we forgot to add it in https://github.com/PyCQA/astroid/pull/1584/

```suggestion
    "Programming Language :: Python :: 3.10",
    "Programming Language :: Python :: 3.11",
```

---

> Should we add `3.11` as well?

We don't support all `3.11` features yet. Especially support for `TryStar` is still missing. Tracking issue #1516
I should probably continue the work on #1389 soon. With that `TryStar` should be fairly strait forward.

---

I think we should add the 3.11 metadata before supporting all 3.11 features, because we're supporting all the old features and if we don't do that, then there's no astroid **at all** for 3.11 which is a lot worse than having astroid without some 3.11 exclusive features.

---

> I think we should add the 3.11 metadata before supporting all 3.11 features, because we're supporting all the old features and if we don't do that, then there's no astroid **at all** for 3.11 which is a lot worse than having astroid without some 3.11 exclusive features.

We don't block anyone from installing astroid with Python 3.11.
However, the way I understand the classifier is "All of the features (or at least the major ones) for version x.xx are supported." That's why I would recommend to wait a bit longer before adding it.